pythonforandroid/python.py

```diff
@@ -113,62 +113,23 @@ def get_recipe_env(self, arch=None, with_flags_in_cc=True):
-        toolchain = '{toolchain_prefix}-{toolchain_version}'.format(
-            toolchain_prefix=self.ctx.toolchain_prefix,
-            toolchain_version=self.ctx.toolchain_version)
-        toolchain = join(self.ctx.ndk_dir, 'toolchains',
-                         toolchain, 'prebuilt', build_platform)
-
-        env['CC'] = (
-            '{clang} -target {target} -gcc-toolchain {toolchain}').format(
-                clang=join(self.ctx.ndk_dir, 'toolchains', 'llvm', 'prebuilt',
-                           build_platform, 'bin', 'clang'),
-                target=arch.target,
-                toolchain=toolchain)
-        env['AR'] = join(toolchain, 'bin', android_host) + '-ar'
-        env['LD'] = join(toolchain, 'bin', android_host) + '-ld'
-        env['RANLIB'] = join(toolchain, 'bin', android_host) + '-ranlib'
-        env['READELF'] = join(toolchain, 'bin', android_host) + '-readelf'
-        env['STRIP'] = join(toolchain, 'bin', android_host) + '-strip'
-        env['STRIP'] += ' --strip-debug --strip-unneeded'
+        env['CC'] = '{}'.format(
+            join(self.ctx.ndk_dir, 'toolchains', 'llvm', 'prebuilt',
+                 build_platform, 'bin', arch.target + '-clang'))
 
         env['PATH'] = (
             '{hostpython_dir}:{old_path}').format(
                 hostpython_dir=self.get_recipe(
                     'host' + self.name, self.ctx).get_path_to_python(),
                 old_path=env['PATH'])
 
-        ndk_flags = (
-            '-fPIC --sysroot={ndk_sysroot} -D__ANDROID_API__={android_api} '
-            '-isystem {ndk_android_host} -I{ndk_include}').format(
-                ndk_sysroot=join(self.ctx.ndk_dir, 'sysroot'),
-                android_api=self.ctx.ndk_api,
-                ndk_android_host=join(
-                    self.ctx.ndk_dir, 'sysroot', 'usr', 'include', android_host),
-                ndk_include=join(self.ctx.ndk_dir, 'sysroot', 'usr', 'include'))
-        sysroot = self.ctx.ndk_platform
-        env['CFLAGS'] = env.get('CFLAGS', '') + ' ' + ndk_flags
-        env['CPPFLAGS'] = env.get('CPPFLAGS', '') + ' ' + ndk_flags
-        env['LDFLAGS'] = env.get('LDFLAGS', '') + ' --sysroot={} -L{}'.format(
-            sysroot, join(sysroot, 'usr', 'lib'))
-
-        # Manually add the libs directory, and copy some object
-        # files to the current directory otherwise they aren't
-        # picked up. This seems necessary because the --sysroot
-        # setting in LDFLAGS is overridden by the other flags.
-        # TODO: Work out why this doesn't happen in the original
-        # bpo-30386 Makefile system.
-        logger.warning('Doing some hacky stuff to link properly')
-        lib_dir = join(sysroot, 'usr', 'lib')
-        if arch.arch == 'x86_64':
-            lib_dir = join(sysroot, 'usr', 'lib64')
-        env['LDFLAGS'] += ' -L{}'.format(lib_dir)
-        shprint(sh.cp, join(lib_dir, 'crtbegin_so.o'), './')
-        shprint(sh.cp, join(lib_dir, 'crtend_so.o'), './')
```

Comment:
@inclement, it seems that we finally can remove this, but why it happened?...your comment about `sysroot` and `LDFLAGS` may be the cause of this...anyway...ndk r19  solves this issue :smile:...so I removed it

Reply:
I think I discovered more recently that the crtbegin_so.o and crtend_so.o issues were due to an NDK bug when using clang as part of the build process, so it makes sense that this is fixed now. The other stuff I never went back to, but certainly happy if it's working now :p